fix(router): remove preserveQueryParams symbol

josephperrott opened the pull request:
Remove preserveQueryParams as it was deprecated for removal in v4, use
queryParamsHandling="preserve" instead.

josephperrott committed:
fix(router): remove preserveQueryParams symbol

Remove preserveQueryParams as it was deprecated for removal in v4, use
queryParamsHandling="preserve" instead.

BREAKING CHANGE: preserveQueryParams has been removed, use
queryParamsHandling="preserve" instead

atscott reviewed (approved):
LGTM, but would add blocked label to wait for the migration PR to merge.

IgorMinar commented on aio/content/guide/deprecations.md:
```diff
@@ -117,13 +116,6 @@ Tip: In the [API reference section](api) of this doc site, deprecated APIs are i
 | --- | ----------- | --------------------- | ----- |
 | [`ngModel` with reactive forms](#ngmodel-reactive) | [`FormControlDirective`](api/forms/FormControlDirective) | v6 | none |
 
-{@a router}
```
instead of deleting this section, can you please move it to the removed apis section: https://angular.io/guide/deprecations#removed-apis-1

IgorMinar replied:
Please update the deprecations.md as mentioned above. thanks!

IgorMinar reviewed (approved):
This one has been deprecated for a long time, so it's great that we are removing it!

Do we understand the impact of this change? it's likely low, but we should verify. I see at least two instances in google3, I haven't tried to search github.

If a migration was part of this PR then this change would be totally fine, but I get that creating migrations for absolutely everything doesn't always makes sense const/benefits-wise. Maybe get @mgechev to weigh in on skipping a migration?

If we are not going to create a migration for this change, then we should at least document the decision in the PR with a brief justification, something along the lines: "This api has been deprecated for a long time, and we don't have evidence that it's being used.

And additionally we should highlight this change in the v11 release notes / update guide.

Otherwise lgtm.

petebacondarwin reviewed (approved):
Reviewed-for: public-api

LGTM after @IgorMinar's comments are addressed.

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

IgorMinar reviewed (changes requested):
just resubmitting https://github.com/angular/angular/pull/38762#pullrequestreview-491047920 to get this out of my review queue for now

Reviewed-for: public-api, docs-packaging-and-releasing

josephperrott committed:
fix(router): create schematic for preserveQueryParams

Create a schematic for migrating preserveQueryParams to use queryParamsHandler
instead.

IgorMinar commented on packages/core/schematics/test/preserve_query_params_migration_spec.ts:
```diff
@@ -0,0 +1,232 @@
+          `const config = { replaceUrl: true, fragment: 'foo', state: {}, queryParamsHandler: 'preserve' };`);
+    });
+
+    it('should migrate when the value is `false`', async () => {
```
```suggestion
    it('should remove when the value is `false`', async () => {
```

IgorMinar commented on packages/core/schematics/test/preserve_query_params_migration_spec.ts:
```diff
@@ -0,0 +1,232 @@
+      expect(content).toContain(`const config = { replaceUrl: true, fragment: 'foo', state: {} };`);
+    });
+
+    it('should migrate when the property is unused', async () => {
```
```suggestion
    it('should not modify when the property is not present', async () => {
```

IgorMinar commented on packages/core/schematics/test/preserve_query_params_migration_spec.ts:
```diff
@@ -0,0 +1,232 @@
+          `this._router.navigate('/', { replaceUrl: true, fragment: 'foo', state: {}, queryParamsHandler: 'preserve' });`);
+    });
+
+    it('should migrate when the value is `false`', async () => {
```
```suggestion
    it('should remove when the value is `false`', async () => {
```

IgorMinar commented on packages/core/schematics/test/preserve_query_params_migration_spec.ts:
```diff
@@ -0,0 +1,232 @@
+          `this._router.createUrlTree(['/'], { replaceUrl: true, fragment: 'foo', state: {} };`);
+    });
+
+    it('should migrate when the property is unused', async () => {
```
```suggestion
    it('should not modify when the property is not present', async () => {
```

IgorMinar commented on packages/core/schematics/migrations/router-preserve-query-params/README.md:
```diff
@@ -0,0 +1,35 @@
+  constructor(private _router: Router) {}
+
+  goHome() {
+    this._router.navigate('/', { skipLocationChange: 'foo', queryParamsHandler: 'preserve' });
```
it is easier to spot the difference if you preserve the prop order:

```suggestion
    this._router.navigate('/', { queryParamsHandler: 'preserve', skipLocationChange: 'foo' });
```

IgorMinar reviewed (changes requested):
just a few nits, the rest looks good.

josephperrott committed:
fixup! fix(router): create schematic for preserveQueryParams

josephperrott committed:
docs: update removed section in deprecations to reference v11

Updates the removed section in the deprecations guide to talk about APIs removed
in v11 rather than the pervious versions.

IgorMinar reviewed (approved):
LGTM! Thanks!

Reviewed-for: public-api

IgorMinar reviewed (approved):
Reviewed-for: global-approvers, docs-packaging-and-releasing, dev-infra,
